### DISCUSSION TITLE
How to have multiple Source tabs?

### DISCUSSION THREAD

**Discussion by littlefengers** [storybookjs/storybook#14583] [Help]
We're migrating our framework to React this year, but we're still going to continue to use our legacy HTML/SCSS framework so we need a way to show code snippets for both.

I've seen this done in other storybooks (https://uikit.wfp.org/docs/index.html?path=/docs/components-ui-elements-button--button-regular).

Does anyone know how this is done or if there's an addon that I could use to do this?

Looking at WFP's source, it looks like they created a custom blocks. Is this the best way to do this or is there an easier way?

https://github.com/wfp/ui/blob/7b0a193c24e7becbca120bec6a5a73de4a17a62b/.storybook/blocks/PurePreview.tsx

Thanks

**Comment by bastiW** [comment]
Same applies for angular components. They usually show HTML, TS Component and the Module.

**Comment by akashyeole** [comment]
How did WFP add the custom multi-tab source (react and html) ?
I also need to implement it the storybook.

**Comment by Loic57** [comment]
Same here, I need it for my web components which don't have the same syntax in angular or react.

**Reply by bastiW** [reply]
For angular you can now use single file components.

We build our custom renderer for our code with two tabs. 

But still would be good to have it in build
